llms/mistral/client_options.go

```diff
@@ -0,0 +1,43 @@
+package mistral
+
+import "time"
+
+type clientOptions struct {
+	apiKey     string
+	endpoint   string
+	maxRetries int
+	timeout    time.Duration
+	model      string
+}
+
+type Option func(*clientOptions)
+
+func WithAPIKey(apiKey string) Option {
```

Comment:
please include comments on exported symbols

Reply:
Will fix 👍

Reply:
I see it's already merged. I'll change these up as a follow-up PR on the coming weekend.